school_attendance/models/school_attendance.py

```diff
@@ -0,0 +1,336 @@
+        "Type",
+        help="Select attendance type",
+    )
+    attendance_sheet_line_ids = fields.One2many(
+        "attendance.sheet.line.matrix", "daily_attendance_id"
+    )
+
+
+class StudentleaveRequest(models.Model):
+    """Defining Model Student Leave Request."""
+
+    _name = "studentleave.request"
+    _description = "Student Leave Request"
+    _inherit = ["mail.thread", "resource.mixin"]
+
+    @api.depends("start_date", "end_date")
+    def _compute_days(self):
+        """Method to compute days based on start-enddate"""
+        for rec in self:
+            if rec.start_date and rec.end_date:
+                rec.days = (rec.end_date - rec.start_date).days + 1
+            if rec.start_date == rec.end_date:
+                rec.days = 1
+            if not rec.start_date or not rec.end_date:
+                rec.days = 0
+
+    name = fields.Char(
+        "Type of Leave", required=True, help="Leave request name"
+    )
+    student_id = fields.Many2one(
+        "student.student", "Student", required=True, help="Enter Student name"
+    )
+    roll_no = fields.Char("Roll Number", help="Enter Student name")
+    standard_id = fields.Many2one(
+        "school.standard", "Class", required=True, help="Select standard"
+    )
+    attachments = fields.Binary("Attachment", help="Attachments")
+    state = fields.Selection(
+        [
+            ("draft", "Draft"),
+            ("toapprove", "To Approve"),
+            ("reject", "Reject"),
+            ("approve", "Approved"),
+        ],
+        "Status",
+        default="draft",
+        tracking=True,
+        help="State of the leave request",
+    )
+    start_date = fields.Date("Start Date", help="Enter start date")
+    end_date = fields.Date("End Date", help="Enter end date")
+    teacher_id = fields.Many2one(
+        "school.teacher", "Class Teacher", help="Select teacher"
+    )
+    days = fields.Integer(
+        "Days", compute="_compute_days", store=True, help="Days of the week"
+    )
+    reason = fields.Text("Reason for Leave", help="Reason for the leave")
+    message_ids = fields.One2many(
+        "mail.message",
+        "res_id",
+        "Messages",
+        domain=lambda self: [("model", "=", self._name)],
+        auto_join=True,
+        help="Message that about to send",
+    )
+    message_follower_ids = fields.One2many(
+        "mail.followers",
+        "res_id",
+        "Followers",
+        domain=lambda self: [("res_model", "=", self._name)],
+        help="Add message followers",
+    )
+
+    @api.onchange("student_id")
+    def onchange_student(self):
+        """Method to get standard and roll no of student selected"""
+        if self.student_id:
+            self.standard_id = self.student_id.standard_id.id
+            self.roll_no = self.student_id.roll_no
+            self.teacher_id = self.student_id.standard_id.user_id.id or False
+
+    def _update_student_vals(self, vals):
```

Comment:
This method have no return {}, Please check

Reply:
This method is used to update the vals in create and write method.
So no need to return anything at last.